integration/tests/observer/execution_state_sync_test.go

```diff
@@ -0,0 +1,237 @@
+package observer
+
+import (
+	"context"
+	"fmt"
+	badgerds "github.com/ipfs/go-ds-badger2"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/onflow/flow-go/engine/ghost/client"
+	"github.com/onflow/flow-go/integration/testnet"
+	"github.com/onflow/flow-go/integration/tests/lib"
+	"github.com/onflow/flow-go/integration/utils"
+	"github.com/onflow/flow-go/model/encoding/cbor"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/state_synchronization"
+	"github.com/onflow/flow-go/network/compressor"
+	storage "github.com/onflow/flow-go/storage/badger"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestExecutionStateSync(t *testing.T) {
+	suite.Run(t, new(ExecutionStateSyncSuite))
+}
+
+type ExecutionStateSyncSuite struct {
+	suite.Suite
+	lib.TestnetStateTracker
+
+	log zerolog.Logger
+
+	bridgeID flow.Identifier
+	ghostID  flow.Identifier
+
+	// root context for the current test
+	ctx            context.Context
+	cancel         context.CancelFunc
+	cancelObserver context.CancelFunc
+
+	net *testnet.FlowNetwork
+}
+
+func (s *ExecutionStateSyncSuite) SetupTest() {
+	logger := unittest.LoggerWithLevel(zerolog.InfoLevel).With().
+		Str("testfile", "execution_state_sync_test.go").
+		Str("testcase", s.T().Name()).
+		Logger()
+
+	s.log = logger
+	s.log.Info().Msg("================> SetupTest")
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	s.buildNetworkConfig()
+
+	// start the network
+	s.net.Start(s.ctx)
+
+	s.startObserver(s.ctx)
+
+	s.Track(s.T(), s.ctx, s.Ghost())
+
+	s.cancelObserver()
+}
+
+func (s *ExecutionStateSyncSuite) TearDownTest() {
+	s.log.Info().Msg("================> Start TearDownTest")
+	//s.cancelObserver()
```

Comment:
Is this unneeded? If so remove, if the cancel is needed, please uncomment. If we want to leave the line commented out please explain.

Reply:
I did some testing. This needs to be reworked a little. The issue is that the observers may be leaked based on testing. I was just conservative not changing the entire test architecture.

Do you agree that the network setup with the observer setup should be part of the test and not the setup code? Moving the observer setup and cleanup into the test would eliminate this comment as well.